hack/podman-net-rule.sh

```
@@ -0,0 +1,218 @@
+  if [[ -z "$subnets" ]]; then
+    echo "ERROR: No subnets found for network '$NET'." >&2
+    exit 1
+  fi
+
+  echo "$subnets" | sed 's/^/  subnet: /'
+
+  while IFS= read -r cidr; do
+    [[ -n "$cidr" ]] || continue
+    # Escape dots in CIDR for regex matching
+    local escaped_cidr="${cidr//./\\.}"
+    if ip rule | grep -qE "^[[:space:]]*$PREF:[[:space:]].*to[[:space:]]+$escaped_cidr[[:space:]].*lookup[[:space:]]+main"; then
```

Shellcheck (0.10.0)

[error] Use braces when expanding arrays, e.g. ${array[idx]} (or ${var}[.. to quiet).

(SC1087)